client/allocdir/task_dir.go

```diff
@@ -0,0 +1,213 @@
+package allocdir
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	cstructs "github.com/hashicorp/nomad/client/structs"
+)
+
+type TaskDir struct {
+	// Dir is the path to Task directory on the host
+	Dir string
+
+	// SharedAllocDir is the path to shared alloc directory on the host
+	// <alloc_dir>/alloc/
+	SharedAllocDir string
+
+	// SharedTaskDir is the path to the shared alloc directory linked into
+	// the task directory on the host.
+	// <task_dir>/alloc/
+	SharedTaskDir string
+
+	// LocalDir is the path to the task's local directory on the host
+	// <task_dir>/local/
+	LocalDir string
+
+	// LogDir is the path to the task's log directory on the host
+	// <alloc_dir>/alloc/logs/
+	LogDir string
+
+	// SecretsDir is the path to secrets/ directory on the host
+	// <task_dir>/secrets/
+	SecretsDir string
+}
+
+// newTaskDir creates a TaskDir struct with paths set. Call Build() to
+// create paths on disk.
+//
+// Call AllocDir.NewTaskDir to create new TaskDirs
+func newTaskDir(allocDir, taskName string) *TaskDir {
```

Comment:
I did that during development and removed it... that probably should have been a hint to keep it!

Fixed.